app/code/core/Mage/Catalog/Model/Resource/Url.php

```diff
@@ -1001,9 +1029,11 @@ protected function _getProducts($productIds, $storeId, $entityId, &$lastEntityId
      *
      * @param int $productId
      * @param int $storeId
+     * @param bool $withDisabled
      * @return Varien_Object|false
+     * @throws Mage_Core_Model_Store_Exception
      */
-    public function getProduct($productId, $storeId)
+    public function getProduct($productId, $storeId, $withDisabled = true)
```

Comment:
@colinmollenhour since we're modifying a public method, we should base it on "next" and consider it a BC no?

Reply:
In my opinion adding an argument with a default value is not a big BC risk since users calling that method in their own code properly should not be affected. Now there could be third-party extensions that already overrode it but it seems very unlikely. So I think "main" is ok in this case.